### ISSUE TITLE
Add supports for `System.Text.Rune` API 

### ISSUE THREAD

**Issue by MangelMaxime** [fable-compiler/Fable#3860]
### Description

For a library I am working on, I will need the [`System.Text.Rune` API](https://learn.microsoft.com/en-us/dotnet/api/system.text.rune?view=net-8.0).

At the moment here is the API I am using:

*`rune.*` are instance methods / properties*

- `Rune.GetRuneAt`
- `rune.ToString()`
- `rune.Utf16SequenceLength`
- `rune.Value`
- `c.EnumerateRunes()` => I could probably use `Rune.GetRuneAt` with a while loop to mimic this API

I think the most difficult part will be to get the `rune.Value` because we need to find how to convert a character/string char into this value. 

Once this is done, adding the different properties / static member should not be too difficult. For JavaScript, [this file](https://github.com/elm/parser/blob/7506b07eaa93a93d13b508b948c016105b0953c8/src/Elm/Kernel/Parser.js) can be a source of inspiration on how to deal with unicode characters especially how to detect the ones using 2 bytes in UTF 16.

```js
// Detect 2 bytes characters
(string.charCodeAt(offset) & 0xF800) === 0xD800
```

I am creating this issue just to keep track of this future work.
